core/src/env/test_env.rs

```diff
@@ -394,24 +394,26 @@ thread_local! {
 }
 
 /// Test environment for testing SRML contract off-chain.
-pub struct TestEnv;
+pub struct TestEnv<T> {
+    marker: PhantomData<T>,
```

Comment:
I always prefer setting this to `marker: PhantomData<fn () -> T>` so that the outer type depends less on `T` and make it `Send` and `Sync`.